### PR TITLE
Restores recent data-table changes

### PR DESCRIPTION
**Pull request by chasestarr**
Reverts uber/baseweb#4801 (which reverted these changes)

### CODE REVIEW

**Review by williamernest — APPROVED** [review]

File: documentation-site/next.config.js [hunk 1]
```diff
@@ -71,6 +72,12 @@ module.exports = withTM(
       webpack: (config, {buildId, dev, isServer, defaultLoaders}) => {
         config.resolve.alias.baseui = resolve(__dirname, '../dist');
         config.resolve.alias.examples = resolve(__dirname, 'examples');
+        // Force a non esm bundle of d3.
+        // Can remove once we upgrade to next 11 or 12 which has support for es modules.
+        config.resolve.alias.d3 = resolve(
```
Comment (williamernest) [inline documentation-site/next.config.js:77]: Next.js doesn't support 3rd part esm modules, have we evaluated this with the latest nextjs?

**Review by williamernest — CHANGES_REQUESTED** [review]

File: documentation-site/next.config.js [hunk 2]
```diff
@@ -71,6 +72,12 @@ module.exports = withTM(
...
+          __dirname,
+          '../node_modules/d3/dist/d3.min.js',
+        );
```
Comment (chasestarr) [inline documentation-site/next.config.js:80]:
```suggestion
```

File: documentation-site/next.config.js [hunk 3]
```diff
@@ -53,6 +53,7 @@ const withTM = require('next-transpile-modules')([
   'gensync',
   'string-width',
   'jsesc',
+  'd3',
```
Comment (chasestarr) [inline documentation-site/next.config.js:56]:
```suggestion
```

File: jest.config.js [hunk 4]
```diff
@@ -22,4 +22,9 @@ module.exports = {
   modulePathIgnorePatterns: ['<rootDir>/dist/'],
   testRegex: '(.|-)test\\.(js|ts|tsx|jsx)$',
   testURL: 'http://localhost/',
+  // d3 uses esm, so we have to manually map to the dist
+  // see https://github.com/facebook/jest/issues/12036
+  moduleNameMapper: {
+    d3: '<rootDir>/node_modules/d3/dist/d3.min.js',
+  },
```
Comment (chasestarr) [inline jest.config.js:29]:
```suggestion
```

**Review by williamernest — APPROVED** [review]

**Pull request merged**